core/domain/exp_services.py

```diff
@@ -1270,3 +1273,44 @@ def search_explorations(query, limit, sort=None, cursor=None):
     """
     return search_services.search(
         query, SEARCH_INDEX_EXPLORATIONS, cursor, limit, sort, ids_only=True)
+
+         
+def _is_suggestion_valid(thread_id, exploration_id): 
+    """Check if the suggestion is still valid. A suggestion is considered 
+    invalid if the exploration version has updated since the suggestion was 
+    made.""" 
+ 
+    states = exp_models.ExplorationModel.get(id=exploration_id).states 
+    suggestion = (feedback_models.SuggestionModel 
+        .get_by_exploration_and_thread_id(exploration_id, thread_id)) 
+    if (suggestion.state_name in states.keys() and 
+           suggestion.state_content == states.get(suggestion.state_name)): 
+        return True 
+    return False        
+
+
+def accept_suggestion(editor_id, change_list, thread_id, exploration_id, 
+                      message):
+    """If the suggestion is valid, accepts it by updating the exploration.
+    Raises an exception if the exploration is not valid."""
+
+    if _is_suggestion_valid(thread_id, exploration_id):
+        commit_message = ('Accepted suggestion by %s: %s' % ( 
+            editor_id, message))
```

Comment:
Did you check if this leads to something human readable?

In fact I think you've got it the other way round: this should use the username (which should be renamed to editor_username), and the update_exploration call should use the editor_id.

What worries me more about this, though, is that this doesn't seem to have been automatically tested. Please could you add a test that uses a nonsensical id and a human-readable name, and ensure that the commit message comes out looking human-readable?